src/Strategy/Composer.hs

```diff
@@ -0,0 +1,139 @@
+  { lockPackages :: [CompDep],
+    lockPackagesDev :: [CompDep]
+  }
+  deriving (Eq, Ord, Show)
+
+data CompDep = CompDep
+  { depName :: Text,
+    depVersion :: Text,
+    depSource :: Source,
+    -- | name to version spec
+    depRequire :: Maybe (Map Text Text),
+    depRequireDev :: Maybe (Map Text Text)
+  }
+  deriving (Eq, Ord, Show)
+
+data Source = Source
+  { sourceType :: Text,
+    sourceUrl :: Text,
+    sourceReference :: Text
+  }
+  deriving (Eq, Ord, Show)
+
+instance FromJSON ComposerLock where
+  parseJSON = withObject "ComposerLock" $ \obj ->
+    ComposerLock <$> obj .: "packages"
+      <*> obj .: "packages-dev"
+
+instance FromJSON CompDep where
+  parseJSON = withObject "CompDep" $ \obj ->
+    CompDep <$> obj .: "name"
+      <*> obj .: "version"
+      <*> obj .: "source"
+      <*> obj .:? "require"
+      <*> obj .:? "require-dev"
+
+instance FromJSON Source where
+  parseJSON = withObject "Source" $ \obj ->
+    Source <$> obj .: "type"
+      <*> obj .: "url"
+      <*> obj .: "reference"
+
+analyze :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs File -> m ProjectClosureBody
+analyze file = mkProjectClosure file <$> readContentsJson @ComposerLock file
+
+mkProjectClosure :: Path Abs File -> ComposerLock -> ProjectClosureBody
+mkProjectClosure file lock =
+  ProjectClosureBody
+    { bodyModuleDir = parent file,
+      bodyDependencies = dependencies,
+      bodyLicenses = []
+    }
+  where
+    dependencies =
+      ProjectDependencies
+        { dependenciesGraph = buildGraph lock,
+          dependenciesOptimal = Optimal,
+          dependenciesComplete = Complete
+        }
+
+newtype CompPkg = CompPkg {pkgName :: Text}
+  deriving (Eq, Ord, Show)
+
+type CompGrapher = LabeledGrapher CompPkg CompLabel
+
+data CompLabel
+  = DepVersion Text
+  | CompEnv DepEnvironment
+  deriving (Eq, Ord, Show)
+
+buildGraph :: ComposerLock -> Graphing Dependency
+buildGraph composerLock = run . withLabeling toDependency $ do
+  traverse_ (addDeps EnvProduction) $ lockPackages composerLock
+  traverse_ (addDeps EnvDevelopment) $ lockPackagesDev composerLock
+  where
+    addDeps :: Has CompGrapher sig m => DepEnvironment -> CompDep -> m ()
+    addDeps env dep = do
+      let pkg = CompPkg (depName dep)
+      _ <- M.traverseWithKey (addEdge pkg) (fromMaybe (M.fromList []) $ depRequire dep)
+      label pkg (DepVersion $ depVersion dep)
+      label pkg (CompEnv env)
+      direct pkg
+
+    addEdge :: Has CompGrapher sig m => CompPkg -> Text -> Text -> m ()
+    addEdge pkg name _ = edge pkg (CompPkg name)
+
+    toDependency :: CompPkg -> Set CompLabel -> Dependency
+    toDependency pkg = foldr addLabel (start pkg)
+
+    addLabel :: CompLabel -> Dependency -> Dependency
+    addLabel (DepVersion ver) dep = dep {dependencyVersion = Just (CEq ver)}
+    addLabel (CompEnv env) dep = dep {dependencyEnvironments = env : dependencyEnvironments dep}
+
+    start :: CompPkg -> Dependency
+    start pkg =
```

Comment:
[nit]: Not sure about the name `start`.  Since `toDependency` is already taken I think this could be merged into `toDependency`:

``` haskell
toDependency pkg = foldr addLabel $
  Dependency
    {  ...
    }
```